### ISSUE TITLE
How can a developer enhance his knowledge from day to day

### ISSUE THREAD

**Issue by slim-hmidi** [kentcdodds/ama#452]
I watched your video mentioned in the youtube and I really appreciate it. However, I have some points to discuss or to ask you about. I follow some famous persons like you on twitter to discover new things and get new information about a technology or library that I didn't know.
Actually I'm a fullstack javascript developer working with react and node.js. In the work I can't learn always new tips due to the kind of tasks to accomplish and due to the deadline of sprint.So the most of time we make copy paste of existing code or adding some features or fixing a bug.
I want to stay up to date with JavaScript (react and node.js) but I didn't know what is the best alternative to follow to enhance my knowledge.
I want to commit in open source project but I'm afraid if I can't succeed in resolving an issue. Sometimes when I read the source code of a library published in Github I didn't understand so many parts of it and I said to myself 'How would you commit in this project and you are not able to understand it'.
I resolve sometimes some problems in hackerrank website but I feel that it's not enough. What do you recommend for me such as solutions and alternatives to follow?

**Comment by kentcdodds** [comment]
Hi! I'm afraid I don't have time to give a super solid answer to all my open questions right now, so I just went through and answered all my AMA questions in the form of a video: https://www.youtube.com/watch?v=4Yq6PO0r-fU&list=PLV5CVI1eNcJgCrPH_e6d57KRUTiDZgs0u

A few things that folks can do to help make things better for others reading the questions and answers:

1. Link to when I start answering the question. Add `&t=xxmxxs` to the end of the URL like this: https://www.youtube.com/watch?v=4Yq6PO0r-fU&list=PLV5CVI1eNcJgCrPH_e6d57KRUTiDZgs0u&t=2m5s
2. Comment with links I referenced
3. Summarize what I talk about

Thanks!

http://kcd.im/what-oss

https://blog.kentcdodds.com/spelunking-in-node-modules-bf165af19968
